src/System.IO.FileSystem.AccessControl/tests/FileSystemAclExtensionsTests.cs

```diff
@@ -224,42 +245,147 @@ public void DirectoryInfo_Create_NullDirectorySecurity()
             }
         }
 
         [Fact]
         public void DirectoryInfo_Create_NotFound()
         {
-            DirectoryInfo info = new DirectoryInfo(@"W:\\I\\Do\\Not\\Exist");
+            DirectoryInfo info = new DirectoryInfo(@"W:\I\Do\Not\Exist");
             DirectorySecurity security = new DirectorySecurity();
-            Assert.Throws<DirectoryNotFoundException>(() => info.Create(security));
+
+            string expectedMessage = $"Could not find a part of the path '{info.FullName}'.";
+            if (PlatformDetection.IsFullFramework)
+            {
+                AssertExtensions.Throws<DirectoryNotFoundException>(() =>
+                    FileSystemAclExtensions.Create(info, security),
+                    expectedMessage);
+            }
+            else
+            {
+                AssertExtensions.Throws<DirectoryNotFoundException>(() =>
+                    info.Create(security),
+                    expectedMessage);
+            }
         }
 
         [Theory]
+        [InlineData(WellKnownSidType.BuiltinUsersSid, FileSystemRights.ReadAndExecute, AccessControlType.Allow)]
+        [InlineData(WellKnownSidType.BuiltinUsersSid, FileSystemRights.ReadAndExecute, AccessControlType.Deny)]
+        [InlineData(WellKnownSidType.BuiltinUsersSid, FileSystemRights.WriteData, AccessControlType.Allow)]
+        [InlineData(WellKnownSidType.BuiltinUsersSid, FileSystemRights.WriteData, AccessControlType.Deny)]
         [InlineData(WellKnownSidType.BuiltinUsersSid, FileSystemRights.FullControl, AccessControlType.Allow)]
-        [InlineData(WellKnownSidType.BuiltinUsersSid, FileSystemRights.ReadData, AccessControlType.Allow)]
-        [InlineData(WellKnownSidType.BuiltinUsersSid, FileSystemRights.Write, AccessControlType.Allow)]
-        [InlineData(WellKnownSidType.BuiltinUsersSid, FileSystemRights.Write, AccessControlType.Deny)]
         [InlineData(WellKnownSidType.BuiltinUsersSid, FileSystemRights.FullControl, AccessControlType.Deny)]
         public void DirectoryInfo_Create_DirectorySecurityWithSpecificAccessRule(
             WellKnownSidType sid,
             FileSystemRights rights,
             AccessControlType controlType)
         {
-
             DirectorySecurity security = GetDirectorySecurity(sid, rights, controlType);
             VerifyDirectorySecurity(security);
         }
 
         #endregion
 
+        #region FileInfo Create
+
+        [Fact]
+        public void FileInfo_Create_NullFileInfo()
+        {
+            FileInfo info = null;
+            FileSecurity security = new FileSecurity();
+
+            if (PlatformDetection.IsFullFramework)
+            {
+                AssertExtensions.Throws<ArgumentNullException>(() =>
+                    FileSystemAclExtensions.Create(info, FileMode.Create, FileSystemRights.WriteData, FileShare.Read, DefaultBufferSize, FileOptions.None, security),
+                    "Value cannot be null.\r\nParameter name: fileInfo");
```

Comment:
Are exceptions localized in .NET Framework? I think so. @ericstj?

Reply:
Yes, if the lang pack is installed.  You shouldn't compare the message.  There should be an assert that is specialized for Argument exceptions for you to compare the paramName is correct.

Reply:
Thanks for the clarification, I wasn't doing it right the first time. I corrected the use whenever possible.